### PR TITLE
Reorder elements to render correctly XSD restrictions and extensions.

### LINKED ISSUE #78: XSD elements order must not be important
Related to #72 PR failing test.

Soapfish generate invalid code if xsd:restriction and xsd:extension are defined before the base element.

I'm trying to fix it, but I don't know the best approach:

1. Reorder XSD elements in runtime.
1. Make `EXTENSION` and `RESTRICTION` classes inherit from `ComplexType` and define the class as a class attribute.
1. Make `EXTENSION` and `RESTRICTION` render after all other types.

All of these options are not easy to implement.

Do you have any other ideas?

I can work on it, but didn't decided how to handle the problem.

### PR DESCRIPTION
**Pull request by iurisilvio**
Fix #78

### CODE REVIEW

File: soapfish/xsd2py.py [hunk 1]
```diff
@@ -116,6 +117,62 @@ def generate_code_from_xsd(xmlelement, known_files=None, location=None,
...
+        else:
+            base = ''
+
+        weights[complex_type.name] = (n, base)
+
+    def _cmp(a, b):
+        try:
+            a = a.name
+        except AttributeError:
+            pass
```
Comment (FelixSchwarz) [inline soapfish/xsd2py.py]: This doesn't look right. If "a" has no attribute "name" you'll get an UnboundLocalError later.

Comment (iurisilvio) [inline soapfish/xsd2py.py]: No. The `a` value is already defined in my cmp arguments. I'm just trying to use the `name` if it is a xsd object. If not, it is a string.

**Comment by FelixSchwarz** [comment]
on a superficial look I think your code does the right thing. I tried some stylistic changes - what do you think about these commits?
https://github.com/FelixSchwarz/soapbase/commit/cc4bc556360672d2dcb4cca5f6dbf551c79d1e5c
https://github.com/FelixSchwarz/soapbase/commit/7da4405b1c2e1a810b0dc366cf74e70ae14a0ee1
https://github.com/FelixSchwarz/soapbase/commit/623fefac78b4c7b3fe0d6e7364669d14140681e6
https://github.com/FelixSchwarz/soapbase/commit/6e59f6671ec37c79062f4ba05da37aa5e848b092

**Comment by iurisilvio** [comment]
I liked your changes. Did just a comment in FelizSchwarz/soapbase@623fefa

**Pull request merged**